cli/popper/commands/cmd_add.py

```diff
@@ -57,20 +58,29 @@ def cli(ctx, pipeline, folder, branch):
         os.mkdir(pipelines_dir)
 
     gh_url = 'https://github.com/{}/{}/'.format(owner, repo)
-    gh_url += 'archive/{}.zip'.format(branch)
+    gh_url += 'archive/{}.tar.gz'.format(branch)
 
     pu.info("Downloading pipeline {}... ".format(pipe_name))
 
-    r = requests.get(gh_url)
+    r = requests.get(gh_url, stream=True)
     if r.status_code != 200:
-        pu.fail("Unable to fetch the pipeline. Please check if the name" +
-                " of the pipeline is correct and the internet is connected")
-    with zipfile.ZipFile(BytesIO(r.content)) as z:
-        z.extractall()
-
-    os.rename('{}-{}/pipelines/{}'.format(repo, branch, pipe_name),
-              os.path.join(folder, pipe_name))
-    shutil.rmtree('{}-{}'.format(repo, branch))
+        pu.fail("Unable to feth the pipeline. Please check if the name" +
+                " of the pipline is correct and the internet is connected")
+
+    file_name = gh_url.split('/')[-1]
+    base_name = os.path.basename(gh_url)
+    with open(file_name, 'wb') as f:
```

Comment:
`BytesIO` does not work? just curious

Reply:
It didn't work here. I was getting an  `invalid file  - <_io.BytesIO object ...` 
error

Reply:
Just to corroborate, if you do `tarfile.TarFile(BytesIO(r.content)) as t:`, you get the `invalid file` message?

Reply:
Yes, I got the invalid file message on this as well. (Just tried this again to double check). 
The complete error :- 
```
File "/home/ujjwal/gsoc/contribute/original/env/lib/python3.5/site-packages/popper/commands/cmd_add.py", line 74, in cli
    with tarfile.TarFile(BytesIO(r.content)) as t:
  File "/home/ujjwal/gsoc/contribute/original/env/lib/python3.5/tarfile.py", line 1423, in __init__
    fileobj = bltn_open(name, self._mode)
TypeError: invalid file: <_io.BytesIO object at 0x7f5d1d841eb8>
```

Reply:
I tried that too. Got the same error :/